samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c

```diff
@@ -656,7 +657,11 @@ static int connect_cloud(void)
 	}
 
 	LOG_INF("Connected to nRF Cloud");
-	return 0;
+	if (IS_ENABLED(CONFIG_NRF_CLOUD_ALERTS)) {
+		err = nrf_cloud_alert_send(ALERT_TYPE_DEVICE_NOW_ONLINE, 0, NULL);
+	}
+
+	return err;
 }
```

Comment:
It's supposed to be a variant on logging.  I'm not sure I understand why you want a callback -- seems overly complex.  I'm about to remove the IS_ENABLED() so it will be just the call.  Would that be acceptable?

Reply:
Even without the guard (which I actually like), I would consider sending an alert like this application-level, and not relevant to maintaining the connection, so my preference would be to separate it out.

Yes, it adds complexity that serves no functional purpose, and yes, it is questionable whether we should approach `connection.c` as though it were a pseudo-library, since it is technically part of the application code for MMSS as a whole, but IMO, it leads to a sample that is easier for beginners to understand.

They won't, for instance, think `nrf_cloud_alert_send` is some important part of connecting. And then all of the device message logic is in one easily understood place (`application.c`)

I just consider alerts to be outside the stated scope of `connection.c`

That being said, if after reading the above you still feel that the added complexity outweighs the loss of separation of concerns I am worried about, then I'll go ahead and approve. Both are relatively minor in the grand scheme of things

Reply:
OK, how about I move it to right after:
`	(void)await_connection(K_FOREVER);`
in application.c: main_application_thread_fn()?

Reply:
I'd be very alright with that! But it would only print on the first connect, not every time re-connection happens. I'm not sure how important that is to you.

Reply:
It's fine.  It's just a demo.